enemies.js

```diff
@@ -76,45 +82,82 @@ class Uoma extends Enemy {
         }
     }
     onCollision(entity) {
-        //super.onCollision(entity);
-        // if (this.state != "DEAD" && entity instanceof CharacterController) { // TODO: check for instanceof CharacterWeapon?
-        //     //entity.dead = true;
-        //     this.state = "DEAD";
-        //     console.log(this.name + " collision with Hornet = LOSS");
-        // }
+        if (this.state != "DEAD" && entity instanceof CharacterController) { // TODO: check for instanceof CharacterWeapon?
+            //entity.dead = true;
+            this.state = "DEAD";
+            console.log(this.name + " collision with Hornet = LOSS");
+        }
         // Need to handle collision with walls?
     }
 };
 class Heavy_Sentry extends Enemy {
     constructor(game, x, y) {
         super(game, x, y);
+
         // TODO: adjust Animator arguments for sprite sheet
         //this.animator = new Animator(this.asset, 4, 22, 172, 148, 6, 0.09, 1, 4);
          //Animator(spritesheet, xStart, yStart, width, height, frameCount, frameDuration,loop, spriteBorderWidth){
+       // console.log(this.asset  )
         this.animationList["IDLE"] = new Animator(this.asset, 4, 325, 208, 263, 6, 0.1, 1, 3 );
-        this.animationList["DEAD"] = new Animator(this.asset, 371, 3600, 154, 167, 3, 0.2, 0, 3);
-        // WALK
+        this.animationList["WALK"] = new Animator(this.asset, 4, 610, 226, 259, 8, 0.15, 1, 3);
+        this.animationList["DEAD"] = new Animator(this.asset, 371, 3600, 154, 164, 3, 0.2, 0, 3);
+        this.animationList["RUN"] = new Animator(this.asset, 4, 1191, 264, 246, 8, 0.1 , 1, 3);
         // ATTACK
         // JUMP
-        this.state = "IDLE";
+        this.state = "RUN";
+        this.updateBB();
     }
     onCollision(entity) {
+        
+        if(entity instanceof Ground && (this.lastBB.bottom-14 <= entity.BB.top)){
+            this.y = entity.BB.top-this.BB.height-14;
+
+        }
         if (entity instanceof CharacterController) {
-            //entity.dead = true;
-            this.state = "DEAD";
+            entity.dead = true;
+            //this.state = "DEAD";
             console.log(this.name + " collision with Hornet = LOSS");
         }
     }
     update() {
+   
         // mechanics for how / where the enemy moves:
         if(this.state == "DEAD") {  // TODO: sound on death?
-            // we don't move on death, and can't do any damage, so no BB.
             this.BB = undefined;
+            this.y = this.y+3;
+
             return;
+        }else{
+            this.x -= (this.speed * this.game.clockTick);
+            this.y += (this.speed * this.game.clockTick);    
         }
         this.updateBB();
         this.collisionChecks();
     }
+    updateBB(){
+        this.lastBB = this.BB;
+        if(this.state == "IDLE" || this.state == "WALK"){
+            this.BB = new BoundingBox(this.game,this.x+40, this.y+20, 140,240, "red");
+        }else if(this.state == "RUN"){
```

Comment:
I know you're a better dev than this. This could be generalized and made better. What's even going on in this block of copypasta?